FileUpload validation code prevents super users from uploading to the default portal

## Description of bug
#3654 introduced a protection mechanism to the FileUpload component (validation code) to make sure the uploads come from a trusted FileUpload component, rendered **to current portalID** and for the current user ID. This protection is good but it currently prevents superusers from uploading files to the default portal (ID = -1) for example, or to any portal different than the one where the FileUpload component was rendered to.
Specifically, this affects the Assets module in Evoq, where the FileUpload component is used to upload files to any portal visible to the user.

## Steps to reproduce
These steps are specific to Evoq, but can apply to other scenarios as well:
1. Open the Assets panel
2. Click on "Global Assets"
3. Upload a file

## Current behavior
"Bad Request" error message is returned.

## Expected behavior
File should get uploaded without issues.

## Screenshots
![image](https://user-images.githubusercontent.com/30123437/85418306-033f3180-b547-11ea-8721-796a811226bf.png)

## Error information
```
2020-06-23 14:46:02.945+00:00 [DNN-QA-01][D:2][T:38][ERROR] DotNetNuke.Web.InternalServices.FileUploadController - System.InvalidOperationException: Bad Request
   at DotNetNuke.Web.InternalServices.FileUploadController.UploadFile(Stream stream, Int32 portalId, UserInfo userInfo, String folder, String filter, String fileName, Boolean overwrite, Boolean isHostPortal, Boolean extract, String validationCode)
```

## Affected version
* [X] 10.00.00 alpha build (in development)
* [X] 09.06.01 latest supported release

## Affected browser
* [X] Chrome
* [X] Firefox
* [X] Safari
* [X] Internet Explorer 11
* [X] Microsoft Edge (Classic)
* [X] Microsoft Edge Chromium

--- a/DNN Platform/DotNetNuke.Web/InternalServices/FileUploadController.cs
+++ b/DNN Platform/DotNetNuke.Web/InternalServices/FileUploadController.cs
@@ -404,7 +404,12 @@ namespace DotNetNuke.Web.InternalServices
                     extensionList = filter.Split(',').Select(i => i.Trim()).ToList();
                 }
 
-                var validateParams = new List<object>{ extensionList, portalId, userInfo.UserID};
+                var validateParams = new List<object>{ extensionList, userInfo.UserID};
+                if (!userInfo.IsSuperUser)
+                {
+                    validateParams.Add(portalId);
+                }
+
                 if (!ValidationUtils.ValidationCodeMatched(validateParams, validationCode))
                 {
                     throw new InvalidOperationException("Bad Request");

--- a/DNN Platform/DotNetNuke.Web/UI/WebControls/DnnFileUploadOptions.cs
+++ b/DNN Platform/DotNetNuke.Web/UI/WebControls/DnnFileUploadOptions.cs
@@ -151,8 +151,11 @@ namespace DotNetNuke.Web.UI.WebControls
                 var parameters = new List<object>(){Extensions};
                 if (portalSettings != null)
                 {
-                    parameters.Add(portalSettings.PortalId);
                     parameters.Add(portalSettings.UserInfo.UserID);
+                    if (!portalSettings.UserInfo.IsSuperUser)
+                    {
+                        parameters.Add(portalSettings.PortalId);
+                    }
                 }
 
                 return ValidationUtils.ComputeValidationCode(parameters);

--- a/DNN Platform/Modules/ResourceManager/View.ascx.cs
+++ b/DNN Platform/Modules/ResourceManager/View.ascx.cs
@@ -67,8 +67,11 @@ namespace Dnn.Modules.ResourceManager
                 if (string.IsNullOrEmpty(_validationCode))
                 {
                     var parameters = new List<object>() { ExtensionWhitelist.Split(',').Select(i => i.Trim()).ToList() };
-                    parameters.Add(PortalSettings.PortalId);
                     parameters.Add(PortalSettings.UserInfo.UserID);
+                    if (!PortalSettings.UserInfo.IsSuperUser)
+                    {
+                        parameters.Add(PortalSettings.PortalId);
+                    }
                     _validationCode = ValidationUtils.ComputeValidationCode(parameters);
                 }
                 return _validationCode;

